python/lsst/meas/base/forcedPhotCcd.py

```diff
@@ -360,20 +448,116 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
 
         tract = butlerQC.quantum.dataId['tract']
         skyMap = inputs.pop("skyMap")
-        inputs['refWcs'] = skyMap[tract].getWcs()
-
-        inputs['refCat'] = self.mergeAndFilterReferences(inputs['exposure'], inputs['refCat'],
-                                                         inputs['refWcs'])
-
-        inputs['measCat'], inputs['exposureId'] = self.generateMeasCat(inputRefs.exposure.dataId,
-                                                                       inputs['exposure'],
-                                                                       inputs['refCat'], inputs['refWcs'],
-                                                                       "visit_detector")
-        self.attachFootprints(inputs['measCat'], inputs['refCat'], inputs['exposure'], inputs['refWcs'])
-        # TODO: apply external calibrations (DM-17062)
-        outputs = self.run(**inputs)
+        refWcs = skyMap[tract].getWcs()
```

Comment:
Only complaint with this runQuantim rewrite, that you changed more than you had to (no lines needed modifying at all), which unnecessarily rejects the `runQuantum` pattern of doing all the disk-touching and input modification (and literally modifying the inputs dict) and then passing the modified `**inputs` to run. You could just *add* these lines between the existing `inputs['refWcs'] = skyMap[tract].getWcs()` and `inputs['refCat'] = self.mergeAndFilterReferences(...`

```   
        # Connections only exist if they are configured to be used.
        skyCorr = inputs.get("skyCorr", None)
        if self.config.useGlobalExternalSkyWcs:
            externalSkyWcsCatalog = inputs.get("externalSkyWcsGlobalCatalog", None)
        else:
            externalSkyWcsCatalog = inputs.get("externalSkyWcsTractCatalog", None)
        if self.config.useGlobalExternalPhotoCalib:
            externalPhotoCalibCatalog = inputs.get("externalPhotoCalibGlobalCatalog", None)
        else:
            externalPhotoCalibCatalog = inputs.get("externalPhotoCalibTractCatalog", None)
        finalizedPsfApCorrCatalog = inputs.get("finalizedPsfApCorrCatalog", None)

        inputs['exposure'] = self.prepareCalibratedExposure(
            inputs["exposure"],
            skyCorr=skyCorr,
            externalSkyWcsCatalog=externalSkyWcsCatalog,
            externalPhotoCalibCatalog=externalPhotoCalibCatalog,
            finalizedPsfApCorrCatalog=finalizedPsfApCorrCatalog
        )
```